node/src/isolate/external.cpp

```diff
@@ -229,11 +233,25 @@ struct external_t::inner_t :
     }
 
     void reconnect(const std::error_code& e) {
-        if(session) {
-            COCAINE_LOG_INFO(log, "reconnecting to external isolation daemon");
-            session->detach(e);
-            session.reset();
-            connect();
+        if(!connecting) {
+            connecting = true;
+            if(session) {
+                session->detach(e);
```

Comment:
JFYI beware of immediate dispatch discard calling during this call (deadlocks, etc).